### PR TITLE
Feature/ds 450 dictionary text

### PR DESCRIPTION
**Pull request by crishpeen**

### CODE REVIEW

File: packages/web-react/src/types/dictionaries.ts [hunk 1]
```diff
@@ -0,0 +1,2 @@
+export type DictionarySizeType = 'small' | 'medium' | 'large';
```
Comment (literat) [inline packages/web-react/src/types/dictionaries.ts:1]: Move this file to `shared/dictionaries.ts`, please.

File: packages/web-react/src/types/dictionaries.ts [hunk 2]
```diff
@@ -0,0 +1,2 @@
...
+export type DictionarySizeExtendedType = 'xsmall' | 'xlarge';
```
Comment (literat) [inline packages/web-react/src/types/dictionaries.ts:2]: Please look at `dropdown.ts` where the `DropdownPlacements` type is defined. We want to define the dictionary the same way.

```ts
export const SIZES = {
  SMALL: 'small'
  // ...
} as const

export type SizesKeys = keyof typeof SIZES;
// using generic for extension
export type Size<E> = typeof SIZES[SizesKeys] | E;

// Now we can use both `small` or Size.SMALL :-)
```

**Review by literat — CHANGES_REQUESTED** [review]

**Comment by literat** [comment]
@crishpeen Just to be sure, there is no updated SCSS, so all classes are generated as we want them?

File: packages/web-react/src/components/Text/stories/TextSizes.tsx [hunk 3]
```diff
@@ -8,6 +8,7 @@ const Story = (props: unknown) => (
     <Text size="large">Body Large Text Regular</Text>
```
Comment (literat) [inline packages/web-react/src/components/Text/stories/TextSizes.tsx:8]: Screenshot 2023-02-01 at 10 02 48
Should you order those sizes in the story, please?

File: DICTIONARIES.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
+# Dictionaries
```
Comment (literat) [inline DICTIONARIES.md:1]: Not sure about the placement of this file. I would prefer something like `/docs/Dictionaries.md`. 

Also not sure whether in the API table of the component should be listed all values of the dictionary for better documentation and usage.

Comment (crishpeen) [reply DICTIONARIES.md:1]: I can move this file, I wasn't sure of the placement either. 

As for the API tables. The idea was, that we will generate variants from types/scss-settings/etc, so when we add a new variant we won't need to update all docs, just one place in the global dictionary. But you can change my mind :)

File: packages/web-react/src/components/Text/stories/TextSizes.tsx (hunk 3 above)
Comment (crishpeen) [reply packages/web-react/src/components/Text/stories/TextSizes.tsx:8]: Should it be sorted by size only in story or everywhere? Currently, we mix ABC and size/scale sorting :(

File: packages/web-react/src/types/dictionaries.ts (hunk 2 above)
Comment (crishpeen) [reply packages/web-react/src/types/dictionaries.ts:2]: ok, i will do it like this

File: packages/web-react/src/types/dictionaries.ts (hunk 1 above)
Comment (crishpeen) [reply packages/web-react/src/types/dictionaries.ts:1]: ok

**Comment by crishpeen** [comment]
> @crishpeen Just to be sure, there is no updated SCSS, so all classes are generated as we want them?

yes, for Text and Heading components, we use helper classes, I think we don't need to change them (it would be huge BC/deprecation) as they work as expected. They basically mirror typography tokens.

**Review by tomassychra — APPROVED** [review]

**Comment by literat** [comment]
@tomassychra @crishpeen I've made an update to this to make my suggestions of types and dictionaries much cleaner. Feel free to review and merge it https://github.com/lmc-eu/spirit-design-system/pull/667

**Review by literat — APPROVED** [review]

**Pull request merged**